core/src/main/scala/kafka/server/AbstractFetcherManager.scala

```diff
@@ -61,7 +61,9 @@ abstract class AbstractFetcherManager[T <: AbstractFetcherThread](val name: Stri
   private[server] def deadThreadCount: Int = lock synchronized { fetcherThreadMap.values.count(_.isThreadFailed) }
 
   def resizeThreadPool(newSize: Int): Unit = {
+    // Used to replace migratePartitions and solve the fetch problem
     def migratePartitions(newSize: Int): Unit = {
+      val allRemovedPartitionsMap = mutable.Map[TopicPartition, InitialFetchState]()
       fetcherThreadMap.forKeyValue { (id, thread) =>
         val partitionStates = removeFetcherForPartitions(thread.partitions)
```

Comment:
While we are here. `removeFetcherForPartitions` iterates overs all the fetchers to remove the `thread.partitions`. That is not necessary because we know that partitions are from a single fetcher. It seems that we could directly use `thread.removePartitions(thread.partitions)` and add `failedPartitions.clear()` before calling `addFetcherForPartitions(allRemovedPartitionsMap)`. What do you think?

Reply:
@yufeiyan1220 Are you interested in doing this in your PR? We can do as a follow-up otherwise.